### DISCUSSION TITLE
Plans to Support ESP32-C6 WiFi 6 TWT (Target Wake Time)?

### DISCUSSION THREAD

**Discussion by itsbeer** [microsoft/devicescript#486] [Q&A] [answered]
Hi Folks,

Does anyone know if there are plans to support the ESP32-C6 WiFi 6 TWT aka Target Wake Time? TWT comes with this new flavor of ESP32 and enables the device to essentially stay connected to the WiFi network but while in sleep.

I guess I have two questions 1) will there be support for the C6 and 2) if there will be, will there be support for TWT?

Espressif has provided an example of code that is supposed to utilize TWT here => https://github.com/espressif/esp-idf/tree/master/examples/wifi/itwt 

Thanks!

**Comment by mmoskal** [comment, marked as answer]
We may add C6 support as more boards become available in the coming months. I did [some work](https://github.com/microsoft/devicescript-esp32/pull/7) on utilizing esp32 sleep modes, but didn't get very far. PRs are always welcome!

**Reply by itsbeer** [reply]
Hi Michal, thanks for the reply. I will take a look. I haven't done much in ESP land. Was hoping the DeviceScript project would be a gateway into it. I just bought a couple of the ESP32-C6 dev kits (ESP32-C6-DEVKITC-1-N8 to be specific) to play around with. DigiKey has 1300+ in stock right now (https://www.digikey.com/short/fr3hbmqh). This being the first ESP that stays connected to WiFi in a sleep state and is supposed to be able to wake up from a network request really really got me interested in it.... 

Keep up the great work! Appreciate it all!
